dali/operators/reader/loader/filesystem.cc

```diff
@@ -13,168 +13,33 @@
 // limitations under the License.
 
 #include "dali/operators/reader/loader/filesystem.h"
-#include <dirent.h>
-#include <errno.h>
-#include <fnmatch.h>
-#include <glob.h>
-#include <sys/stat.h>
-#include <algorithm>
-#include <cstring>
-#include <string>
-#include <utility>
-#include <vector>
-#include "dali/core/call_at_exit.h"
-#include "dali/core/error_handling.h"
-#include "dali/operators/reader/loader/utils.h"
 
 namespace dali {
 namespace filesystem {
 
-inline bool starts_with(const std::string &str, const char *prefix) {
-  // TODO(janton): this is a substitute for C++20's string::starts_with
-  // trick: this only matches if the prefix is found at the beginning of the string
-  return str.rfind(prefix, 0) == 0;
-}
-
 std::string join_path(const std::string &dir, const std::string &path) {
   if (dir.empty())
     return path;
   if (path.empty())
     return dir;
-  if (path[0] == dir_sep)  // absolute path
+
+  char separator = dir_sep;
+#ifdef WINVER
+  // If an URI, use slash as a separator
+  if (URI::Parse(dir).valid)
+    separator = '/';
+#endif
+
+  if (path[0] == separator)  // absolute path
```

Comment:
This is actually a bug (?):
```
dir = "s3://bucket/"
path  = "/dataset"
join_path(dir, path) == "/dataset"  <- oops, we've lost the scheme and authority.
```

Reply:
path would be `dataset/` in any case, because we get relative paths to to the file_root. So I think this should not happen

Reply:
Well, you can just type them in in `files` argument.